### DISCUSSION TITLE
Camera resets when changing src

### DISCUSSION THREAD

**Discussion by designbyadrian** [google/model-viewer#3290] [Q&A]
### Description

**Steps to reproduce:**

1. rotate and zoom a model
2. using JavaScript, set the `src` attribute to another GLB

**Expected result:**

Model is updated, with the camera at the same position as before

**Actual result:**

Camera is reset, zooming out to the initial on load.

#### Live Demo

Please see below:

[https://joyous-plain-offer.glitch.me](https://joyous-plain-offer.glitch.me)

### Version

- model-viewer: v1.10.1

### Browser Affected
- [ x ] Chrome, version: latest
- [ x ] Edge
- [ x ] Firefox
- [ x ] IE
- [ x ] Safari

### OS

- [ x ] Android
- [ x ] iOS
- [ x ] Linux
- [ x ] MacOS
- [ x ] Windows

**Comment by elalish** [comment]
This is intended behavior, especially since different models might be at vastly different scales, so if the camera didn't move it could end up inside the new object, or looking at blank white. If you know your models all have the same bounding radius, then it shouldn't be too hard to use `getCameraOrbit`, `getCameraTarget`, getFieldOfView` and then set those when the new model loads.

**Reply by designbyadrian** [reply]
Thank you for a prompt reply.

My issue is that I'm trying to set the camera as soon as the load event fires, but then I experience race conditions.

**Reply by elalish** [reply]
If you can post a Glitch of that, we might be able to help. Generally when working with `Lit` like our component, using `await element.updateComplete;` tends to help a lot of with race conditions. See our tests for more examples.

**Reply by designbyadrian** [reply]
Well, it turns out that most of the race conditions are on my side, but I still need to juggle the camera changes with the simples example, especially with how the camera moves into position. There is no camera-move-end (or similar), so I need to debounce camera-change. Please see new example: https://funny-holistic-empress.glitch.me/

I'd prefer if the camera reset on source-change would be an option you could toggle off.

**Reply by elalish** [reply]
I think you can skip the camera-change event and just use `getCameraOrbit` and such when the user selects a different model.

**Reply by DanieleMagri** [reply]
How did you update the model?
